drivers/acpi/acpi.c

```diff
@@ -0,0 +1,543 @@
+					  NULL);
+	if (ACPI_FAILURE(status)) {
+		ACPI_EXCEPTION((AE_INFO, status, "While installing Notify handler"));
+		goto exit;
+	}
+
+	status = AcpiInstallAddressSpaceHandler(ACPI_ROOT_OBJECT, ACPI_ADR_SPACE_SYSTEM_MEMORY,
+						region_handler, region_init, NULL);
+	if (ACPI_FAILURE(status)) {
+		ACPI_EXCEPTION((AE_INFO, status, "While installing an OpRegion handler"));
+	}
+exit:
+
+	return status;
+}
+
+static ACPI_STATUS initialize_acpica(void)
+{
+	ACPI_STATUS status;
+
+	/* Initialize the ACPI subsystem */
+	status = AcpiInitializeSubsystem();
+	if (ACPI_FAILURE(status)) {
+		ACPI_EXCEPTION((AE_INFO, status, "While initializing ACPI"));
+		goto exit;
+	}
+
+	/* Initialize the ACPI Table Manager and get all ACPI tables */
+	status = AcpiInitializeTables(NULL, 16, FALSE);
+	if (ACPI_FAILURE(status)) {
+		ACPI_EXCEPTION((AE_INFO, status, "While initializing Table Manager"));
+		goto exit;
+	}
+
+	/* Initialize the ACPI hardware */
+	status = AcpiEnableSubsystem(ACPI_FULL_INITIALIZATION);
+	if (ACPI_FAILURE(status)) {
+		ACPI_EXCEPTION((AE_INFO, status, "While enabling ACPI"));
+		goto exit;
+	}
+
+	/* Install local handlers */
+	status = install_handlers();
+	if (ACPI_FAILURE(status)) {
+		ACPI_EXCEPTION((AE_INFO, status, "While installing handlers"));
+		goto exit;
+	}
+
+	/* Create the ACPI namespace from ACPI tables */
+	status = AcpiLoadTables();
+	if (ACPI_FAILURE(status)) {
+		ACPI_EXCEPTION((AE_INFO, status, "While loading ACPI tables"));
+		goto exit;
+	}
+
+	/* Complete the ACPI namespace object initialization */
+	status = AcpiInitializeObjects(ACPI_FULL_INITIALIZATION);
+	if (ACPI_FAILURE(status)) {
+		ACPI_EXCEPTION((AE_INFO, status, "While initializing ACPI objects"));
+	}
+exit:
+
+	return status;
+}
+
+static ACPI_NAMESPACE_NODE *acpi_name_Lookup(char *name)
+{
+	char *path;
+	ACPI_STATUS status;
+	ACPI_NAMESPACE_NODE *node = NULL;
+
+	LOG_DBG("");
+
+	status = AcpiNsInternalizeName(name, &path);
+	if (ACPI_FAILURE(status)) {
+		LOG_ERR("Invalid namestring: %s", name);
+		return NULL;
+	}
+
+	status = AcpiNsLookup(NULL, path, ACPI_TYPE_ANY, ACPI_IMODE_EXECUTE,
+			      ACPI_NS_NO_UPSEARCH | ACPI_NS_DONT_OPEN_SCOPE, NULL, &node);
+	if (ACPI_FAILURE(status)) {
+		LOG_ERR("Could not locate name: %s, %d", name, status);
+	}
+
+	ACPI_FREE(path);
+	return node;
+}
+
+static ACPI_NAMESPACE_NODE *acpi_evaluate_method(char *bus_name, char *method)
+{
+	ACPI_NAMESPACE_NODE *node;
+	ACPI_NAMESPACE_NODE *handle;
+	ACPI_NAMESPACE_NODE *prt_node = NULL;
+
+	LOG_DBG("%s", bus_name);
+
+	handle = acpi_name_Lookup(bus_name);
+	if (!handle) {
+		LOG_ERR("No ACPI node with given name: %s", bus_name);
+		goto exit;
+	}
+
+	if (handle->Type != ACPI_TYPE_DEVICE) {
+		LOG_ERR("No ACPI node foud with given name: %s", bus_name);
+		goto exit;
+	}
+
+	node = ACPI_CAST_PTR(ACPI_NAMESPACE_NODE, handle);
+
+	(void)AcpiGetHandle(node, method, ACPI_CAST_PTR(ACPI_HANDLE, &prt_node));
```

Comment:
I don't think it is really needed to call AcpiEvaluateObject() for AcpiGetIrqRoutingTable(), AcpiRsGetCrsMethodData() etc. since it already invoke internal function AcpiUtEvaluateObject ().  Anyway I will check with ACPICA team on this.

Reply:
Then why is this function called "acpi_evaluate_method" if it's not evaluating anything?  Name it "acpi_get_bus_method" or somesuch?